node/channel.js

```diff
@@ -319,6 +319,11 @@ TChannel.prototype.listen = function listen(port, host, callback) {
 TChannel.prototype.register = function register(name, options, handler) {
     var self = this;
 
+    // If we are the top channel
+    if (!self.topChannel) {
```

Comment:
Apparently my comment on this didn't get submitted. If this logic is correct, it reads extremely counter-intuitively to the comment.